compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.cpp

```diff
@@ -363,6 +363,90 @@ CompilationInfoAttr::verify(function_ref<InFlightDiagnostic()> emitError,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// iree_codegen.workgroup_mapping
+//===----------------------------------------------------------------------===//
+
+WorkgroupMappingAttr WorkgroupMappingAttr::get(MLIRContext *context,
+                                               WorkgroupId id) {
+  return WorkgroupMappingAttr::get(context, id, /*delinearizedDim=*/0);
+}
+
+LogicalResult
+WorkgroupMappingAttr::verify(function_ref<InFlightDiagnostic()> emitError,
+                             WorkgroupId id, int64_t delinearizedDim) {
+  if (delinearizedDim > 0 && id != WorkgroupId::IdZ) {
+    return emitError() << "illegal to use `delinearizationDim` for "
+                       << stringifyWorkgroupId(id);
+  }
+  return success();
+}
+
+LogicalResult WorkgroupMappingAttr::verifyAttrList(
+    MLIRContext *context, function_ref<InFlightDiagnostic()> emitError,
+    ArrayRef<Attribute> attrs) {
+  if (attrs.empty()) {
```

Comment:
What does an empty mapping list mean?

Reply:
its a corner case, and I was trying to think about whether I should fail or pass. For now I am just ignoring but it shouldnt happen. 
Really I cant test the list anywhere. Subsequent PRs will use this method. Ill revisit if empty list needs to be handled differently.